txn_writer.go

```diff
@@ -316,13 +321,13 @@ func (sw *SafeWriter) Write(root string, sm gps.SourceManager, noExamples bool)
 
 		// 0666 is before umask; mirrors behavior of os.Create (used by
 		// writeFile())
-		if err = ioutil.WriteFile(filepath.Join(td, ManifestName), append([]byte(initOutput), tb...), 0666); err != nil {
+		if err := writeFile(filepath.Join(td, ManifestName), sw.Manifest, initOutput); err != nil {
```

Comment:
Instead of converting the `HasManifest()` path to rely on `writeFile()` (which has the duplicate marshaling issue @sectioneight notes) like `HasLock(), let's make the `HasLock()` path follow the `HasManifest()` path pattern - marshal once initially, then `ioutil.WriteFile()` with everything all at once.

Then, we can drop just `writeFile()`, as it'll be dead code. Please do that, too :D